bindgen/TypeTranslator.cpp

```diff
@@ -68,111 +57,143 @@ TypeTranslator::TranslateFunctionPointer(const clang::QualType &qtpe,
+
+    auto it = aliasesMap.find(name);
+    if (it != aliasesMap.end()) {
+        /* name contains space: enum <name>.
+         * Use type alias instead enum type */
+        return (*it).second;
+    }
+    /* type has typedef alias */
+    return ir.getTypeDefWithName(name);
+}
+
+Type *TypeTranslator::translateConstantArray(const clang::ConstantArrayType *ar,
+                                             const std::string *avoid) {
     const uint64_t size = ar->getSize().getZExtValue();
-    const std::string nat = uint64ToScalaNat(size);
-    return "native.CArray[" + Translate(ar->getElementType(), avoid) + ", " +
-           nat + "]";
+    return new ArrayType(translate(ar->getElementType(), avoid), size);
 }
 
-std::string TypeTranslator::Translate(const clang::QualType &qtpe,
-                                      const std::string *avoid) {
+Type *TypeTranslator::translate(const clang::QualType &qtpe,
+                                const std::string *avoid) {
 
     const clang::Type *tpe = qtpe.getTypePtr();
 
     if (typeEquals(tpe, avoid)) {
         // This is a type that we want to avoid the usage.
-        //Êxample: A struct that has a pointer to itself
+        // Êxample: A struct that has a pointer to itself
         uint64_t size = ctx->getTypeSize(tpe);
-        return "native.CArray[Byte, " + uint64ToScalaNat(size) + "]";
+        return new ArrayType(new PrimitiveType("Byte"), size);
     }
 
     if (tpe->isFunctionPointerType()) {
-        return TranslateFunctionPointer(qtpe, avoid);
+        return translateFunctionPointer(qtpe, avoid);
 
     } else if (tpe->isPointerType()) {
         return TranslatePointer(
             tpe->getAs<clang::PointerType>()->getPointeeType(), avoid);
 
-    } else if (qtpe->isStructureType() || qtpe->isUnionType()) {
-        return handleReservedWords(TranslateStructOrUnion(qtpe));
+    } else if (qtpe->isStructureType()) {
+        return translateStruct(qtpe);
+
+    } else if (qtpe->isUnionType()) {
+        return translateUnion(qtpe);
 
     } else if (qtpe->isEnumeralType()) {
-        return TranslateEnum(qtpe);
+        return translateEnum(qtpe);
 
     } else if (qtpe->isConstantArrayType()) {
-        return TranslateConstantArray(ctx->getAsConstantArrayType(qtpe), avoid);
+        return translateConstantArray(ctx->getAsConstantArrayType(qtpe), avoid);
     } else if (qtpe->isArrayType()) {
         return TranslatePointer(ctx->getAsArrayType(qtpe)->getElementType(),
                                 avoid);
     } else {
 
         auto found = typeMap.find(qtpe.getUnqualifiedType().getAsString());
         if (found != typeMap.end()) {
-            return handleReservedWords(found->second);
+            return new PrimitiveType(found->second);
         } else {
-            // TODO: Properly handle non-default types
-            return handleReservedWords(qtpe.getUnqualifiedType().getAsString());
+            return ir.getTypeDefWithName(
```

Comment:
I guess here we don't want to go through the alias map?

Reply:
No, because `struct myStruct` will be translated above (the same for `enum` and `union`)